src/components/vm/nics/vmNicsCard.jsx

```diff
@@ -105,6 +107,95 @@ VmNetworkActions.propTypes = {
     networks: PropTypes.array.isRequired,
 };
 
+const NetworkSource = ({ network, networkId, vm, hostDevices }) => {
+    const id = vmId(vm.name);
+    const checkDeviceAvailability = (network) => {
+        for (const i in hostDevices) {
+            if (hostDevices[i].valid && hostDevices[i].Interface == network) {
+                return true;
+            }
+        }
+        return false;
+    };
+
+    const sourceJump = (source) => {
+        return () => {
+            if (source !== null && checkDeviceAvailability(source)) {
+                cockpit.jump(`/network#/${source}`, cockpit.transport.host);
+            }
+        };
+    };
+
+    const singleSourceElem = source => {
+        let label = rephraseUI("networkType", network.type);
+        label = label.charAt(0).toUpperCase() + label.slice(1);
+
+        return (
+            <DescriptionListGroup>
+                <DescriptionListTerm>
+                    {label}
+                </DescriptionListTerm>
+                <DescriptionListDescription>
+                    <Flex spaceItems={{ default: 'spaceItemsSm' }} alignItems={{ default: 'alignItemsCenter' }} id={`${id}-network-${networkId}-source`}>
+                        <FlexItem>
+                            {checkDeviceAvailability(source)
+                                ? <Button variant="link" isInline onClick={sourceJump(source)}>{source}</Button>
+                                : source}
+                        </FlexItem>
+                        {needsShutdownIfaceSource(vm, network) && <NeedsShutdownTooltip iconId={`${id}-network-${networkId}-source-tooltip`} tooltipId="tip-network" />}
+                    </Flex>
+                </DescriptionListDescription>
+            </DescriptionListGroup>
+        );
+    };
+    const addressPortSourceElem = source => (<>
+        <DescriptionListGroup>
+            <DescriptionListTerm>
+                {_("Address")}
+            </DescriptionListTerm>
+            <DescriptionListDescription id={`${id}-network-${networkId}-address`}>
+                {source.address}
+            </DescriptionListDescription>
+        </DescriptionListGroup>
+        <DescriptionListGroup>
+            <DescriptionListTerm>
+                {_("Port")}
+            </DescriptionListTerm>
+            <DescriptionListDescription id={`${id}-network-${networkId}-port`}>
+                {source.port}
+            </DescriptionListDescription>
+        </DescriptionListGroup>
```

Comment:
These 17 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1077-20230626-135722-bc6440ae-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)